fuzz/targets/mod.rs

```diff
@@ -0,0 +1,212 @@
+    let n = cursor.read_as_u32()?;
+    let mut buf = vec![];
+    let _ = buf.encode_u32(n);
+    let _ = buf.encode_u32_le(n);
+    Ok(())
+}
+
+fn fuzz_codec_number_u32_decode(data: &[u8]) -> Result<(), Error> {
+    let buf = data.to_owned();
+    let _ = tikv::util::codec::number::decode_u32(&mut buf.as_slice());
+    let _ = tikv::util::codec::number::decode_u32_le(&mut buf.as_slice());
+    Ok(())
+}
+
+fn fuzz_codec_number_i32_encode(data: &[u8]) -> Result<(), Error> {
+    use tikv::util::codec::number::NumberEncoder;
+    let mut cursor = Cursor::new(data);
+    let n = cursor.read_as_i32()?;
+    let mut buf = vec![];
+    let _ = buf.encode_i32_le(n);
+    Ok(())
+}
+
+fn fuzz_codec_number_i32_decode(data: &[u8]) -> Result<(), Error> {
+    let buf = data.to_owned();
+    let _ = tikv::util::codec::number::decode_i32_le(&mut buf.as_slice());
+    Ok(())
+}
+
+fn fuzz_codec_number_u16_encode(data: &[u8]) -> Result<(), Error> {
+    use tikv::util::codec::number::NumberEncoder;
+    let mut cursor = Cursor::new(data);
+    let n = cursor.read_as_u16()?;
+    let mut buf = vec![];
+    let _ = buf.encode_u16(n);
+    let _ = buf.encode_u16_le(n);
+    Ok(())
+}
+
+fn fuzz_codec_number_u16_decode(data: &[u8]) -> Result<(), Error> {
+    let buf = data.to_owned();
+    let _ = tikv::util::codec::number::decode_u16(&mut buf.as_slice());
+    let _ = tikv::util::codec::number::decode_u16_le(&mut buf.as_slice());
+    Ok(())
+}
+
+fn fuzz_coprocessor_codec_decimal(data: &[u8]) -> Result<(), Error> {
+    use tikv::coprocessor::codec::mysql::decimal::{Decimal, RoundMode};
+
+    fn fuzz(lhs: &Decimal, rhs: &Decimal, cursor: &mut Cursor<&[u8]>) -> Result<(), Error> {
+        let _ = lhs.clone().abs();
+        let _ = lhs.ceil();
+        let _ = lhs.floor();
+        let _ = lhs.prec_and_frac();
+
+        let mode = match cursor.read_as_u8()? % 3 {
+            0 => RoundMode::HalfEven,
+            1 => RoundMode::Truncate,
+            _ => RoundMode::Ceiling,
+        };
+        let frac = cursor.read_as_i8()?;
+        let _ = lhs.clone().round(frac, mode.clone());
+
+        let shift = cursor.read_as_u64()? as isize;
+        let _ = lhs.clone().shift(shift);
+
+        let _ = lhs.as_i64();
+        let _ = lhs.as_u64();
+        let _ = lhs.as_f64();
+        let _ = lhs.is_zero();
+        let _ = lhs.approximate_encoded_size();
+
+        let _ = lhs > rhs;
+
+        let _ = lhs + rhs;
+        let _ = lhs - rhs;
+        let _ = lhs * rhs;
+        let _ = lhs.clone() / rhs.clone();
+        let _ = lhs.clone() % rhs.clone();
+        let _ = -lhs.clone();
+        Ok(())
+    }
+
+    let mut cursor = Cursor::new(data);
+    let decimal1 = Decimal::from_f64(cursor.read_as_f64()?)?;
+    let decimal2 = Decimal::from_f64(cursor.read_as_f64()?)?;
+    let _ = fuzz(&decimal1, &decimal2, &mut cursor);
+    let _ = fuzz(&decimal2, &decimal1, &mut cursor);
+    Ok(())
+}
+
+register_fuzz!(
+    fuzz_codec_bytes_encode_bytes,
```

Comment:
Could you seperate different series of functions to different binarys? By this way, fuzzer can find better inputs.